### DISCUSSION TITLE
'DataFrame' object has no attribute 'startswith'

### DISCUSSION THREAD

**Discussion by juanxo90** [opengeos/leafmap#100] [Q&A] [answered]
Hello everyone

I have this error, I really do not kave any idea why it happen. After I cleanen my data using pandas, I got this error when I tried to add to the map. I don't know what I did worng when I create my dataframe.

**Comment by giswqs** [comment, marked as answer]
Please provide the source code that can reproduce the error.

**Comment by juanxo90** [comment]
I write this code  @giswqs 

```
m = leafmap.Map(center=[19.5744, -98.8441], zoom=4)

m.add_xy_data(pz, x="LATITUD", y="LONGITUD", layer_name="Piezometria")

m
```

and I got this error: 
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-65-db9a853a2269> in <module>()
      3 
      4 # Agregamos los puntos desde el pandas dataframe
----> 5 m.add_xy_data(pz, x="LATITUD", y="LONGITUD", layer_name="Piezometria")
      6 
      7 # Visualizamos de forma similar a leafmap

1 frames
/usr/local/lib/python3.7/dist-packages/pandas/core/generic.py in __getattr__(self, name)
   5139             if self._info_axis._can_hold_identifiers_and_holds_name(name):
   5140                 return self[name]
-> 5141             return object.__getattribute__(self, name)
   5142 
   5143     def __setattr__(self, name: str, value) -> None:

AttributeError: 'DataFrame' object has no attribute 'startswith'



The `pz` is a pandas dataframe with different well data, and latitude and longitude in WGS84

**Comment by juanxo90** [comment]
The problem is solved now, I forgot to use the geopandas dataframe

**Reply by arabinda90** [reply]
I am also facing the same problem can you please provide the exact code so that I can plot my shape file to the leafmap?

My code was

import os
import leafmap
import glob
import geopandas as gpd

df = gpd.read_file("D:/F/Shape File/wb/S_bengal/S_bengal_1.shp")

m = leafmap.Map()
m
m.add_shp(df, layer_name="S_bengal_1")

# the error is -----------------------------------
AttributeError                            Traceback (most recent call last)
Input In [69], in <cell line: 5>()
      3 m = leafmap.Map()
      4 m
----> 5 m.add_shp(df, layer_name="S_bengal_1")

File ~\anaconda3\envs\gdal_env\lib\site-packages\leafmap\leafmap.py:2167, in Map.add_shp(self, in_shp, layer_name, style, hover_style, style_callback, fill_colors, info_mode, zoom_to_layer, encoding)
   2148 """Adds a shapefile to the map.
   2149 
   2150 Args:
   (...)
   2162     FileNotFoundError: The provided shapefile could not be found.
   2163 """
   2165 import glob
-> 2167 if in_shp.startswith("http") and in_shp.endswith(".zip"):
   2168     out_dir = os.path.abspath("./cache/shp")
   2169     if not os.path.exists(out_dir):

File ~\anaconda3\envs\gdal_env\lib\site-packages\pandas\core\generic.py:5575, in NDFrame.__getattr__(self, name)
   5568 if (
   5569     name not in self._internal_names_set
   5570     and name not in self._metadata
   5571     and name not in self._accessors
   5572     and self._info_axis._can_hold_identifiers_and_holds_name(name)
   5573 ):
   5574     return self[name]
-> 5575 return object.__getattribute__(self, name)

AttributeError: 'GeoDataFrame' object has no attribute 'startswith'

**Reply by arabinda90** [reply]
okay I got my result 

shapefile_path = "D:/F/Shape File/wb/S_bengal/S_bengal_1.shp"
m = leafmap.Map()
m.add_shp(shapefile_path)
m
